test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc

```diff
@@ -0,0 +1,97 @@
+
+#include "extensions/filters/network/common/utility.h"
+#include "extensions/filters/network/well_known_names.h"
+
+#include "test/extensions/filters/common/ext_authz/test_common.h"
+#include "test/extensions/filters/network/common/fuzz/uber_readfilter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+
+std::vector<absl::string_view> UberFilterFuzzer::filterNames() {
+  // These filters have already been covered by this fuzzer.
+  // Will extend to cover other network filters one by one.
+  static ::std::vector<absl::string_view> filter_names_;
+  if (filter_names_.empty()) {
+    filter_names_ = {NetworkFilterNames::get().ExtAuthorization,
+                     NetworkFilterNames::get().LocalRateLimit,
+                     NetworkFilterNames::get().RedisProxy,
+                     NetworkFilterNames::get().ClientSslAuth,
+                     NetworkFilterNames::get().Echo,
+                     NetworkFilterNames::get().DirectResponse,
+                     NetworkFilterNames::get().DubboProxy,
+                     NetworkFilterNames::get().SniCluster};
+  }
+  return filter_names_;
+}
+
+void UberFilterFuzzer::perFilterSetup(const std::string& filter_name) {
+  // Set up response for ext_authz filter
+  if (filter_name == NetworkFilterNames::get().ExtAuthorization) {
+
+    async_client_factory_ = std::make_unique<Grpc::MockAsyncClientFactory>();
+    async_client_ = std::make_unique<Grpc::MockAsyncClient>();
+    // TODO(jianwendong): consider testing on different kinds of responses.
+    ON_CALL(*async_client_, sendRaw(_, _, _, _, _, _))
+        .WillByDefault(testing::WithArgs<3>(Invoke([&](Grpc::RawAsyncRequestCallbacks& callbacks) {
+          Filters::Common::ExtAuthz::GrpcClientImpl* grpc_client_impl =
+              dynamic_cast<Filters::Common::ExtAuthz::GrpcClientImpl*>(&callbacks);
+          const std::string empty_body{};
+          const auto expected_headers =
+              Filters::Common::ExtAuthz::TestCommon::makeHeaderValueOption({});
+          auto check_response = Filters::Common::ExtAuthz::TestCommon::makeCheckResponse(
+              Grpc::Status::WellKnownGrpcStatus::Ok, envoy::type::v3::OK, empty_body,
+              expected_headers);
+          // Give response to the grpc_client by calling onSuccess().
+          grpc_client_impl->onSuccess(std::move(check_response), span_);
+          return async_request_.get();
+        })));
+
+    EXPECT_CALL(*async_client_factory_, create()).WillOnce(Invoke([&] {
+      return std::move(async_client_);
+    }));
+
+    EXPECT_CALL(factory_context_.cluster_manager_.async_client_manager_,
+                factoryForGrpcService(_, _, _))
+        .WillOnce(Invoke([&](const envoy::config::core::v3::GrpcService&, Stats::Scope&, bool) {
+          return std::move(async_client_factory_);
+        }));
+  }
+}
+
+void UberFilterFuzzer::checkInvalidInputForFuzzer(const std::string& filter_name,
```

Comment:
I think that any fuzzer whose name is not in {DirectResponse, LocalRateLimit} has no constraint on its config (for now).  It is worth explaining this in a comment.

Reply:
Added! Thanks!